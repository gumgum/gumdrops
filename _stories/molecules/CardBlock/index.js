import React from 'react';
import { text, object } from '@storybook/addon-knobs';
import { optionalSelect } from '../../../components/utils/optionalSelect';

import readme from './README.md';
import Card from '../../../components/molecules/Card';
import CardBlock from '../../../components/molecules/CardBlock';

const options = {
    'divide-top': 'divide-top',
    'divide-bottom': 'divide-bottom',
    '': 'No Value'
};

const component = () => (
    <Card>
        <CardBlock
            option={optionalSelect('Option', options, '')}
            className={text('ClassName', '')}
            style={object('Style', {})}>
            <p>Content in a CardBlock.</p>
        </CardBlock>
    </Card>
);

export default [readme, component];
