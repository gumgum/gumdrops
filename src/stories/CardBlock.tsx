import React from 'react';
import cx from 'classnames';

export interface CardBlockProps extends React.HTMLAttributes<Element> {
    option?: 'divide-top' | 'divide-bottom';
}

const baseClass = 'gds-card__block';

export const CardBlock: React.FC<CardBlockProps> = ({
    option,
    className,
    children,
    ...otherProps
}) => {
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${option}`]: option
    });

    return (
        <div className={rootClass} {...otherProps}>
            {children}
        </div>
    );
};
