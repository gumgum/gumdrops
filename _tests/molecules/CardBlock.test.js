import renderer from 'react-test-renderer';
import React from 'react';
import CardBlock from '../../components/molecules/CardBlock';

const defaultProps = {
    option: 'divide-top',
    className: 'custom',
    style: { width: 120 }
};

test('Expect <CardBlock> to render properly', () => {
    const tree = renderer
        .create(
            <CardBlock {...defaultProps}>
                <div>Foo</div>
            </CardBlock>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
