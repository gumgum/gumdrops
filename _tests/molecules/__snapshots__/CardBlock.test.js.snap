// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <CardBlock> to render properly 1`] = `
<div
  className="gds-card__block custom gds-card__block--divide-top"
  style={
    {
      "width": 120,
    }
  }
>
  <div>
    Foo
  </div>
</div>
`;
